pandas/core/arrays/integer.py

```diff
@@ -604,12 +618,17 @@ def value_counts(self, dropna=True):
             # w/o passing the dtype
             array = np.append(array, [self._mask.sum()])
             index = Index(
-                np.concatenate([index.values, np.array([np.nan], dtype=object)]),
+                np.concatenate(
+                    [index.values, np.array([self.dtype.na_value], dtype=object)]
+                ),
                 dtype=object,
             )
 
         return Series(array, index=index)
 
+    def _values_for_factorize(self) -> Tuple[np.ndarray, Any]:
+        return self._coerce_to_ndarray(na_value=np.nan), np.nan
```

Comment:
This preserves the behavior / implementation on master.

I think someday we may want masked-versions of our algos, but that's down the road.

Reply:
> I think someday we may want masked-versions of our algos, but that's down the road.

Yes, I was thinking the same in the BooleanArray PR. Created an issue fr this: https://github.com/pandas-dev/pandas/issues/30037